src/Input/NumberInput.tsx

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import styled from "styled-components";
+
+import { colors } from "../shared/constants";
+
+const StyledContainer = styled.div`
+  width: 100%;
+  position: relative;
+  overflow-x: hidden;
+`;
+
+const StyledInner = styled.div`
+  width: 100%;
+  display: inline;
```

Comment:
Why is this `inline`? I'll add more comments below but let's use flexbox for layouts.
Having this
```css
display: flex;
flex-direction: column;
justify-content: space-between;
```
And its children not have `float` will likely give you the same expected layout.

Good resources for flexbox include:
- https://css-tricks.com/snippets/css/a-guide-to-flexbox/
- https://www.youtube.com/watch?v=K74l26pE4YA